cyclonedx/model/bom.py

```diff
@@ -280,7 +280,7 @@ def metadata(self) -> BomMetaData:
         return self._metadata
 
     @metadata.setter
-    def metadata(self, metadata: BomMetaData) -> None:
+    def metadata(self, metadata: Optional[BomMetaData]) -> None:
```

Comment:
why? 
MEtadata is just an data-class.
If it was empty, we could just omit it from rendering/serializing when it comes to it.
Why should we have it optional?

if it was optional, then the downstream implementations would be horrible - as you always would need to check if metadata is there. Ans like you showed with the already done implementations in <https://github.com/CycloneDX/cyclonedx-python-lib/pull/327/files#diff-9c84a99c5332704dd6a90a4defadfd9df437336f07b79b91033e0a96bbd8fbf1> and others ... 
```py
if not bom.metadata:
  bom.metatada = new BomMetadata() 
bom.metadata.timestamp = now
bom.metadata.whatsoever = something
...
```

What is the point in making the models match the JSON 1:1 ? 
The JSON is just one representation of the models. just like the XML is, and the ProtoBuff is. 

---

The models are a general purpose thing with a design. 
The JSON/XML/PB are bound to the constraints of their specific systems.